kubeflow-pipelines.yaml

```diff
@@ -0,0 +1,178 @@
+            cp -r ../samples/* ${{targets.subpkgdir}}/samples
+            cp ./src/$path/config/sample_config.json ${{targets.subpkgdir}}/samples/sample_config.json
+          fi
+
+  - name: "kubeflow-pipelines-cache-deployer"
+    description: "Kubeflow Pipelines cache-deployer"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/usr/bin
+          mkdir -p ${{targets.subpkgdir}}/third_party
+          install -m755 -D third_party/license.txt ${{targets.subpkgdir}}/third_party/license.txt
+          install -m755 -D backend/src/cache/deployer/deploy-cache-service.sh ${{targets.subpkgdir}}/usr/bin/deploy-cache-service.sh
+          install -m755 -D backend/src/cache/deployer/webhook-create-signed-cert.sh ${{targets.subpkgdir}}/usr/bin/webhook-create-signed-cert.sh
+          install -m755 -D backend/src/cache/deployer/webhook-patch-ca-bundle.sh ${{targets.subpkgdir}}/usr/bin/webhook-patch-ca-bundle.sh
+
+  - name: "kubeflow-pipelines-cache-deployer-compat"
+    dependencies:
+      runtime:
+        - kubeflow-pipelines-cache-deployer
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/kfp/cache/deployer/
+          ln -s /usr/bin/deploy-cache-service.sh "${{targets.subpkgdir}}"/kfp/cache/deployer/deploy-cache-service.sh
+          ln -s /usr/bin/webhook-create-signed-cert.sh "${{targets.subpkgdir}}"/kfp/cache/deployer/webhook-create-signed-cert.sh
+          ln -s /usr/bin/webhook-patch-ca-bundle.sh "${{targets.subpkgdir}}"/kfp/cache/deployer/webhook-patch-ca-bundle.sh
+
+  - name: "kubeflow-pipelines-frontend-compat"
+    description: "Kubeflow Pipelines frontend"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/server
+          mkdir -p ${{targets.subpkgdir}}/client
+          cd frontend/
+          npm ci
+          npm run postinstall
+          npm run build
+          cp -r ./server/* ${{targets.subpkgdir}}/server
+          cp -r ./build/* ${{targets.subpkgdir}}/client
+
+  - name: "kubeflow-pipelines-metadata-writer"
+    description: "Kubeflow Pipelines metadata_writer"
+    dependencies:
+      runtime:
+        - python3
+        - py3-absl-py
+        - py3-attrs
+        - py3-cachetools
+        - py3-certifi
+        - py3-charset-normalizer
+        - py3-google-auth
+        - py3-grpcio
+        - py3-idna
+        - py3-kubernetes
+        - py3-lru-dict
+        - py3-ml-metadata
+        - py3-oauthlib
+        - py3-protobuf
+        - py3-asn1-modules
+        - py3-asn1
+        - py3-dateutil
+        - py3-yaml
+        - py3-requests-oauthlib
+        - py3-requests
+        - py3-rsa
+        - py3-six
+        - py3-urllib3
+        - py3-websocket-client
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/opt/metadata_writer
+          cd backend/metadata_writer
+          pip3 install -r requirements.txt --ignore-installed --prefix=/usr --root="${{targets.destdir}}"
```

Comment:
Is this the right way to use? since all deps has been included in `runtime`.

Reply:
To be sure I wonder if we need `--find-links /usr/bin`or something?

https://stackoverflow.com/questions/11091623/how-to-install-packages-offline

Reply:
Actually, I'm not familiar with that flag, and it seems no actively used in the codebase 🤔

Reply:
It would be good to confirm we are using the local packages and not downloading them here.  Do the build logs show anything?

Also I wonder if @strongjz or @kaniini know more?

Reply:
We can check the APK to see if any unexpected files are there

Reply:
All the site-packages are matching with what included in the `dependencies.runtime`: https://pastebin.com/rxThunQL

Reply:
I removed this line because we install all the packages, we should not need to do "pip install" again.

Reply:
migrated `pip3 install` to `gpep517`.